src/scripts/main.js

```diff
@@ -1,3 +1,217 @@
+        messageWin.classList.remove('hidden');
+      };
+    }
+  }
+};
+
+function hasEmptyCell() {
+  for (let row = 0; row < 4; row++) {
+    for (let cell = 0; cell < 4; cell++) {
+      if (gameField[row][cell] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function canMove() {
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 3; c++) {
+      if (gameField[c][r] === gameField[c + 1][r]
+        || gameField[r][c] === gameField[r][c + 1]) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function setTwoOrFour() {
+  canMove();
+
+  if (!canMove() && !hasEmptyCell()) {
+    messageLose.classList.remove('hidden');
+
+    return;
+  }
+
+  const row = Math.floor(Math.random() * 4);
+  const cell = Math.floor(Math.random() * 4);
+
+  if (hasEmptyCell()) {
+    (gameField[row][cell] === 0)
+      ? gameField[row][cell] = Math.floor(Math.random() >= 0.9 ? 4 : 2)
+      : setTwoOrFour();
+  }
+
+  setGameField();
+}
+
+function filteringZero(row) {
+  return row.filter(num => num !== 0);
+}
+
+function slide(row) {
+  let filteredRow = filteringZero(row);
+
+  for (let i = 0; i < filteredRow.length - 1; i++) {
+    if (filteredRow[i] === filteredRow[i + 1]) {
+      filteredRow[i] *= 2;
+      filteredRow[i + 1] = 0;
+      score += filteredRow[i];
+    }
+  }
+  filteredRow = filteringZero(filteredRow);
+
+  while (filteredRow.length < 4) {
+    filteredRow.push(0);
+  };
+
+  return filteredRow;
+}
+
+function slideLeft() {
+  for (let r = 0; r < gameField.length; r++) {
+    let row = gameField[r];
+
+    row = slide(row);
+    gameField[r] = row;
+  }
+
+  setGameField();
+}
+
+function slideRight() {
+  for (let r = 0; r < gameField.length; r++) {
+    let row = gameField[r].reverse();
+
+    row = slide(row);
+    gameField[r] = row.reverse();
+  }
+
+  setGameField();
+}
+
+function slideUp() {
+  for (let c = 0; c < gameField.length; c++) {
+    let column = [
+      gameField[0][c],
+      gameField[1][c],
+      gameField[2][c],
+      gameField[3][c],
+    ];
+
+    column = slide(column);
+
+    for (let r = 0; r < gameField.length; r++) {
+      gameField[r][c] = column[r];
+    }
+  }
+
+  setGameField();
+}
+
+function slideDown() {
+  for (let c = 0; c < gameField.length; c++) {
+    let column = [
+      gameField[0][c],
+      gameField[1][c],
+      gameField[2][c],
+      gameField[3][c],
+    ];
+
+    column.reverse();
+    column = slide(column);
+    column.reverse();
+
+    for (let r = 0; r < gameField.length; r++) {
+      gameField[r][c] = column[r];
+    }
+  }
+
+  setGameField();
+}
+
+button.addEventListener('click', () => {
+  if (button.textContent === 'Start') {
+    button.textContent = 'Restart';
+    button.classList.replace('start', 'restart');
+    messeageStart.classList.add('hidden');
+    messageWin.classList.add('hidden');
+    messageLose.classList.add('hidden');
+    setTwoOrFour();
+    setTwoOrFour();
```

Comment:
I recommend you to move this to separate functions like startGame() and restartGame().